docs/swagger-doc-coverage.md

```diff
@@ -0,0 +1,178 @@
+objects than mentioned here, you can find the complete details at
+[Swagger/OpenAPI specification page](http://swagger.io/specification).
+
+## Working with the `zulip.yaml` file
+
+A Swagger specification file has three general parts: information and configuration,
+endpoint definitions, and object schemas referenced by other objects (as an alternative
+to defining everything inline.)
+
+### Configuration
+
+These objects, at the top of `zulip.yaml`, identify the API, define the backend host
+for the working examples, list supported schemes and types of authentication, and
+configure other settings. Once defined, information in this section rarely changes.
+
+For example, the `swagger` and `info` objects look like this:
+```
+# Basic Swagger UI info
+swagger: '2.0'
+info:
+  version: '1.0.0'
+  title: Zulip REST API
+  description: Powerful open source group chat
+  contact:
+    url: https://zulip.org/
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0.html
+```
+
+### Endpoint definitions
+
+The Paths Object contains [Path Item Objects](http://swagger.io/specification/#pathItemObject)
+for each endpoint path, which describe in detail the methods and parameters it
+accepts and the responses it returns.
+
+There is one Path Item Object for each supported method, containing a
+[Parameters Object](http://swagger.io/specification/#parameterObject) describing
+the required and optional inputs. A
+[Response Object](http://swagger.io/specification/#responseObject) similarly specifies
+the content of the response. They may reference schemas from a global Definitions
+Object (see [Schemas](#schemas), below.)
+
+Example:
+
+The `/messages` endpoint (shown here simplified) accepts a POST with four
+parameters (one optional), expects HTTP Basic authentication, and returns a JSON
+response of type `MessageResponse`. The parameter type `formData` means parameters
+are expected as POST data, as you might do with `curl -d`.
+
+```
+paths:
+
+  /messages:
+    post:
+      description: Send a message
+      operationId: addMessage
+      produces:
+        - application/json
+      parameters:
+        - name: type
+          in: formData
+          description: One of {`private`, `stream`}
+          required: true
+          type: string
+        - name: content
+          in: formData
+          description: Content
+          required: true
+          type: string
+        - name: to
+          in: formData
+          description: Recipient
+          required: true
+          type: string
+        - name: subject
+          in: formData
+          description: Topic
+          required: false
+          type: string
+      security:
+        - basicAuth: []
+      responses:
+        '200':
+          description: Response
+          schema:
+            $ref: '#/definitions/MessageResponse'
+        default:
+          description: Unexpected error
+          schema:
+            $ref: '#/definitions/ErrorModel'
+```
```

Comment:
I would consider adding adding a first example of something that doesn't require arguments at all, e.g. `GET /users`, just to show what a really simple one looks like.